### ISSUE TITLE
Synchronization issue on cuda all cuda implementations

### ISSUE THREAD

**Issue by F-Madruga** [koallen/parallel-floyd-warshall#1]
I think you have synchronization issue because your thrying to read from k_j_value from the matrix while writting in it

**Comment by koallen** [comment]
Yeah I think you are right, this code was written years ago for a course project. If you wanna make a PR, feel free to do so :) Thanks for bringing it up though.
